File: CodingChallenges/CC_06_Mitosis.py/Cell.py
# Daniel Shiffman - Translated  to Python by Ali Al Saqban
# http://codingtra.in
# http://patreon.com/codingtrain

# Mitosis
# https://youtu.be/jxGS3fKPKJA

class Cell(object):
    def __init__(self, pos=None,r=None, c=None):
        self.pos = pos.copy() if pos else PVector(random(width),random(height))
        self.r = r if r else 60
        self.c = c if c else color(random(100,255),0,random(100,255),100)
    
    def clicked(self, x, y):
        d = dist(self.pos.x, self.pos.y, x, y)
        return d < self.r
    
    def mitosis(self):
        return Cell(self.pos,self.r*0.8,self.c)
    
    def move(self):
        vel = PVector.random2D()
        self.pos.add(vel)
    
    def show(self):
        noStroke()
        fill(self.c)
        ellipse(self.pos.x,self.pos.y,self.r,self.r)

File: CodingChallenges/CC_006_Mitosis/P5/cell.js
// Daniel Shiffman
// http://codingtra.in
// http://patreon.com/codingtrain
// Code for: https://youtu.be/jxGS3fKPKJA

function Cell(pos, r, c) {
  if (pos) {
    this.pos = pos.copy();
  } else {
    this.pos = createVector(random(width), random(height));
  }

  this.r = r || 60;
  this.c = c || color(random(100, 255), 0, random(100, 255), 100);

  this.clicked = function(x, y) {
    var d = dist(this.pos.x, this.pos.y, x, y);
    if (d < this.r) {
      return true;
    } else {
      return false;
    }
  };

  this.mitosis = function() {
    //this.pos.x += random(-this.r, this.r);
    var cell = new Cell(this.pos, this.r * 0.8, this.c);
    return cell;
  };

  this.move = function() {
    var vel = p5.Vector.random2D();
    this.pos.add(vel);
  };

  this.show = function() {
    noStroke();
    fill(this.c);
    ellipse(this.pos.x, this.pos.y, this.r, this.r);
  };
}
